src/Symfony/Component/Translation/Formatter/FallbackFormatter.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Formatter;
+
+use Symfony\Component\Translation\Exception\LogicException;
+
+class FallbackFormatter implements MessageFormatterInterface, ChoiceMessageFormatterInterface
+{
+    /**
+     * @var MessageFormatterInterface|ChoiceMessageFormatterInterface
+     */
+    private $firstFormatter;
+
+    /**
+     * @var MessageFormatterInterface|ChoiceMessageFormatterInterface
+     */
+    private $secondFormatter;
+
+    public function __construct(MessageFormatterInterface $firstFormatter, MessageFormatterInterface $secondFormatter)
+    {
+        $this->firstFormatter = $firstFormatter;
+        $this->secondFormatter = $secondFormatter;
+    }
+
+    public function format($message, $locale, array $parameters = array())
+    {
+        try {
+            $result = $this->firstFormatter->format($message, $locale, $parameters);
+        } catch (\Throwable $e) {
```

Comment:
Can we be more specific here? We should be as much as possible IMHO (same below.)

Reply:
Do we want to be more specific?
I know the IntlMesaageFormatter only throws InvalidArgument, but isn’t it a good idea to catch everything to give the second formatter a change to run?